hail_search/queries/mito.py

```diff
@@ -112,6 +118,215 @@ class MitoHailTableQuery(BaseHailTableQuery):
+        merged_comp_het_ht = self._merge_project_hts(filtered_comp_het_hts, n_partitions)
+        return merged_ht, merged_comp_het_ht
+
+    @staticmethod
+    def _build_family_index_map(sorted_wes_family_sample_data, sorted_wgs_family_sample_data):
+        family_guid_idx_map = defaultdict(dict)
+
+        for family_idx, samples in enumerate(sorted_wes_family_sample_data):
+            family_guid = samples[0]['familyGuid']
+            family_guid_idx_map[family_guid][SAMPLE_TYPE_WES] = family_idx
+
+        for family_idx, samples in enumerate(sorted_wgs_family_sample_data):
+            family_guid = samples[0]['familyGuid']
+            family_guid_idx_map[family_guid][SAMPLE_TYPE_WGS] = family_idx
+
+        return hl.literal(family_guid_idx_map)
+
+    def _annotate_passes_quality_filter(self, ht: hl.Table, quality_filter, **kwargs) -> hl.Table:
+        wes_passes_quality_filter = self._get_family_passes_quality_filter(quality_filter, ht, WES_FILTERS_FIELD, **kwargs)
+        wes_passes = ht[WES_FAMILY_ENTRIES_FIELD] if wes_passes_quality_filter is None else (
+            ht[WES_FAMILY_ENTRIES_FIELD].map(lambda entries: hl.or_missing(wes_passes_quality_filter(entries), entries))
+        )
+        ht = ht.annotate(wes_passes_quality=wes_passes)
+
+        wgs_passes_quality_filter = self._get_family_passes_quality_filter(quality_filter, ht, WGS_FILTERS_FIELD, **kwargs)
+        wgs_passes = ht[WGS_FAMILY_ENTRIES_FIELD] if wgs_passes_quality_filter is None else (
+            ht[WGS_FAMILY_ENTRIES_FIELD].map(lambda entries: hl.or_missing(wgs_passes_quality_filter(entries), entries))
+        )
+        ht = ht.annotate(wgs_passes_quality=wgs_passes)
+        return ht
+
+    @staticmethod
+    def _apply_filters(ht, family_idx_map) -> hl.Table:
+        # Keep family from both sample types if either passes quality and either passes inheritance
+        ht = ht.annotate(
+            wes_family_entries=hl.enumerate(ht[WES_FAMILY_ENTRIES_FIELD]).map(
+                lambda x: hl.or_missing(  # x[0] is family index, x[1] is list of samples
+                    (hl.is_defined(ht.wes_passes_quality[x[0]]) | hl.if_else(
+                        family_idx_map.get(x[1][0]['familyGuid'], hl.empty_dict(hl.tstr, hl.tint32)).contains(SAMPLE_TYPE_WGS),
+                        hl.is_defined(ht.wgs_passes_quality[family_idx_map[x[1][0]['familyGuid']][SAMPLE_TYPE_WGS]]),
+                        False
+                    )) & (hl.is_defined(ht.wes_passes_inheritance[x[0]]) | hl.if_else(
+                        family_idx_map.get(x[1][0]['familyGuid'], hl.empty_dict(hl.tstr, hl.tint32)).contains(SAMPLE_TYPE_WGS),
+                        hl.is_defined(ht.wgs_passes_inheritance[family_idx_map[x[1][0]['familyGuid']][SAMPLE_TYPE_WGS]]),
+                        False
+                    )), x[1]
+            )),
+            wgs_family_entries=hl.enumerate(ht[WGS_FAMILY_ENTRIES_FIELD]).map(
```

Comment:
this looks very similar to the above `wes_family_entries` annotation, might be nice to have a shared helper function

Reply:
I added a shared helper to reduce some code duplication